services/core/shared/core.js

```diff
@@ -355,6 +460,78 @@ const EMAcalc = (feePerByte, prevFeeEstPerByte) => {
 	return EMAoutput;
 };
 
+const getEstimateFeeByte = async () => {
+	const cacheKeyFeeEst = 'lastFeeEstimate';
+
+	const prevFeeEstPerByte = { blockHeight: config.feeEstimates.defaultStartBlockHeight };
+	const cachedFeeEstPerByte = await cacheRedisFees.get(cacheKeyFeeEst);
+	const latestBlock = await getBlocks({ sort: 'height:desc', limit: 1 });
+	if (cachedFeeEstPerByte
+		&& ['low', 'med', 'high', 'updated', 'blockHeight', 'blockId']
+			.every(key => Object.keys(cachedFeeEstPerByte).includes(key))) {
+		if ((Date.now() / 1000) - cachedFeeEstPerByte.updated < 10
+			|| Number(latestBlock.data.id) === cachedFeeEstPerByte.blockHeight) {
+			return cachedFeeEstPerByte;
+		}
+
+		Object.assign(prevFeeEstPerByte, cachedFeeEstPerByte);
+	}
+
+	const coreLogic = async (blockBatch, innerPrevFeeEstPerByte) => {
+		blockBatch.data = await BluebirdPromise.map(blockBatch.data, async block => Object
+			.assign(block, { transactions: await getTransactions({ blockId: block.id }) }),
+			{ concurrency: blockBatch.data.length });
+
+		const wavgBlockBatch = calculateWeightedAvg(blockBatch.data);
+		const sizeLastBlock = calculateBlockSize(blockBatch.data[0]);
+		const feePerByte = calculateFeePerByte(blockBatch.data[0]);
+		const innerFeeEstPerByte = {};
+
+		if (wavgBlockBatch > (12.5 * 2 ** 10) || sizeLastBlock > (14.8 * 2 ** 10)) {
+			const EMAoutput = EMAcalc(feePerByte, innerPrevFeeEstPerByte);
+
+			innerFeeEstPerByte.low = EMAoutput.feeEstLow;
+			innerFeeEstPerByte.med = EMAoutput.feeEstMed;
+			innerFeeEstPerByte.high = EMAoutput.feeEstHigh;
+		} else {
+			innerFeeEstPerByte.low = 0;
+			innerFeeEstPerByte.med = 0;
+			innerFeeEstPerByte.high = 0;
+		}
+
+		innerFeeEstPerByte.updated = Math.floor(Date.now() / 1000);
+		innerFeeEstPerByte.blockHeight = blockBatch.data[0].height;
+		innerFeeEstPerByte.blockId = blockBatch.data[0].id;
+
+		return innerFeeEstPerByte;
+	};
+
+	const range = size => Array(size).fill().map((_, index) => index);
+	const feeEstPerByte = {};
+	const blockBatch = {};
+	do {
```

Comment:
if the previous calculation exists, shouldn't it use that as an offset and generate the next estimate?
it looks like the previous calculation is not used?

Reply:
We currently do use the previous calculation (as stated in [LIP-0016](prevFeeEstPerByte)) to generate the next estimate. In the code excerpt below (line 518), `prevFeeEstPerByte` does refer to the previous estimation that is initially retrieved from the cache (kindly refer to the second code block, lines 466-477):

```
Object.assign(prevFeeEstPerByte, await coreLogic(blockBatch, prevFeeEstPerByte));
```

```
const prevFeeEstPerByte = { blockHeight: config.feeEstimates.defaultStartBlockHeight };
const cachedFeeEstPerByte = await cacheRedisFees.get(cacheKeyFeeEst);
const latestBlock = await getBlocks({ sort: 'height:desc', limit: 1 });
if (cachedFeeEstPerByte
	&& ['low', 'med', 'high', 'updated', 'blockHeight', 'blockId']
		.every(key => Object.keys(cachedFeeEstPerByte).includes(key))) {
	if ((Date.now() / 1000) - cachedFeeEstPerByte.updated < 10
		|| Number(latestBlock.data.id) === cachedFeeEstPerByte.blockHeight) {
		return cachedFeeEstPerByte;
	}

	Object.assign(prevFeeEstPerByte, cachedFeeEstPerByte);
}
```